src/diff/props.js

```diff
@@ -9,27 +8,32 @@ import { assign } from '../util';
  * @param {object} newProps The new props
  * @param {object} oldProps The old props
  * @param {boolean} isSvg Whether or not this node is an SVG node
+ * @param {boolean} hydrate Whether or not we are in hydration mode
  */
-export function diffProps(dom, newProps, oldProps, isSvg) {
+export function diffProps(dom, newProps, oldProps, isSvg, hydrate) {
 	let i;
 
-	const keys = Object.keys(newProps).sort();
-	for (i = 0; i < keys.length; i++) {
-		const k = keys[i];
-		if (k!=='children' && k!=='key' && (!oldProps || ((k==='value' || k==='checked') ? dom : oldProps)[k]!==newProps[k])) {
-			setProperty(dom, k, newProps[k], oldProps[k], isSvg);
+	for (i in oldProps) {
+		if (!(i in newProps)) {
+			setProperty(dom, i, null, oldProps[i], isSvg);
 		}
 	}
 
-	for (i in oldProps) {
-		if (i!=='children' && i!=='key' && !(i in newProps)) {
-			setProperty(dom, i, null, oldProps[i], isSvg);
+	for (i in newProps) {
+		if ((!hydrate || typeof newProps[i]=='function') && i!=='value' && i!=='checked' && oldProps[i]!==newProps[i]) {
```

Comment:
@marvinhagemeister idea: check for `i[0]=='o' && i[1]=='n'` instead? it should be ~free.

Reply:
Tried that but it's `+8 B` bigger :S